src/Streamly/Internal/LZ4/Config.hs

```diff
@@ -115,6 +116,13 @@ data BlockSize =
     | BlockMax1MB
     | BlockMax4MB
 
+blockSizeToInt :: BlockSize -> Int
+blockSizeToInt BlockHasSize = maxBound
```

Comment:
We can restrict this to maxUncompLen.

Reply:
Move this function to LZ4.hs  where it is actually used.

Reply:
This behaves like an instance almost? Only one right answer for this. Having this locally with the type makes more sense to me.

Reply:
I'll move it anyway. (Dependency issue)

Reply:
If this function is used by multiple modules then it makes sense to keep it in a common module. Otherwise I like to keep a function close to where it is used. Closely related code goes together.